app/models/mixins/relationship_mixin.rb

```diff
@@ -527,10 +581,14 @@ def child_types(*args)
   end
 
   def add_parent(parent)
+    return parent.add_children(self) if use_ancestry?
+
     parent.with_relationship_type(relationship_type) { parent.add_child(self) }
   end
 
   def add_children(*child_objs)
+    return child_objs.each { |child| child.with_relationship_type(relationship_type) { child.parent=(self) } } if use_ancestry?
```

Comment:
sigh

does it work with `update_attribute`?
I agree with you that you need to use the with relationship type gook here
since this is inside the file that defined it, I'm ok with that.

Reply:
that blowing out of the cache when this is changed. not sure if we need to do that but I'm leaning towards yes. (not condoning, just trying to follow the existing code to keep the same behavior/bugs)

Reply:
not marking as resolved because of the second comment though I did follow suggestion to use `update!`